tests/Game.class.test.js

```diff
@@ -0,0 +1,724 @@
+      });
+
+      it('should merge cells with the same value to the left and update the score', () => {
+        const game2048 = new Game(initialStateMergingSingle, 0, 'playing');
+
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        expect(state[0][0]).toBe(4);
+        expect(state[1][0]).toBe(4);
+        expect(state[2][0]).toBe(4);
+        expect(state[3][0]).toBe(4);
+
+        expect(game2048.getScore()).toBe(16);
+      });
+
+      it('should merge cells with the same value to the left multiple times if possible and update the score', () => {
+        const game2048 = new Game(initialStateMergingMultiple, 0, 'playing');
+
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        expect(state[0][0]).toBe(4);
+        expect(state[0][1]).toBe(8);
+        expect(state[1][0]).toBe(4);
+        expect(state[1][1]).toBe(8);
+        expect(state[2][0]).toBe(16);
+        expect(state[2][1]).toBe(4);
+        expect(state[3][0]).toBe(16);
+        expect(state[3][1]).toBe(4);
+
+        expect(game2048.getScore()).toBe(64);
+      });
+
+      it('should merge cells with the same value and stack with those that are not merged to the left and update the score', () => {
+        const game2048 = new Game(initialStateMergingAndStacking, 0, 'playing');
+
+        game2048.moveLeft();
+
+        const state = game2048.getState();
+
+        expect(state[0][0]).toBe(4);
+        expect(state[0][1]).toBe(2);
+        expect(state[1][0]).toBe(4);
+        expect(state[2][0]).toBe(4);
+        expect(state[2][1]).toBe(4);
+        expect(state[3][0]).toBe(4);
+        expect(state[3][1]).toBe(2);
+
+        expect(game2048.getScore()).toBe(12);
+      });
+    });
+
+    describe('moveRight', () => {
+      it('should not move the board if the game is not in the "playing" state', () => {
+        const game2048 = new Game(initialStateNoStacking);
+
+        game2048.moveRight();
+
+        const state = game2048.getState();
+
+        expect(state).toEqual(initialStateNoStacking);
+      });
+
+      it('should move all non-empty cells to the right and don\'t change the score', () => {
+        const game2048 = new Game(initialStateNoStacking, 0, 'playing');
+
+        game2048.moveRight();
+
+        const state = game2048.getState();
+
+        expect(state[0][3]).toBe(2);
+        expect(state[1][3]).toBe(4);
+        expect(state[2][3]).toBe(8);
+        expect(state[3][3]).toBe(16);
+
+        expect(game2048.getScore()).toBe(0);
+      });
+
+      it('should add a random cell with 2 or 4 at an empty position after the move', () => {
+        const game2048 = new Game(initialStateNoStacking, 0, 'playing');
+
+        game2048.moveRight();
+
+        const flatState = game2048.getState().flat();
+
+        expect(
+          flatState
+            .some((cell, index) => (
+              index % 4 !== 3 && [2, 4].includes(cell)
+            )),
+        ).toBe(true);
```

Comment:
I think that we can test the numbers of not empty cells 
and the total sum of numbers 

then we can just check that 1 non-empty cell was added and total sum increase <= 4